Source/DafnyCore/Backends/Rust/RustBackend.cs

```diff
@@ -170,23 +169,41 @@ private string ComputeExeName(string targetFilename) {
         await cargoTomlWriter.WriteLineAsync();
       }
     }
+  }
 
-    var args = new List<string> {
-      "build",
-      "--quiet"
-    };
-
-    if (callToMain == null) {
-      args.Add("--lib");
-    } else {
-      args.Add("--bin");
-      args.Add(Path.GetFileNameWithoutExtension(targetFilename));
+  private static void ImportRustRuntimeTo(string targetDirectory) {
```

Comment:
Would it be possible to use `SinglePassCodeGenerator.EmitRuntimeSource` or `.ReadRuntimeSystem` instead, perhaps with some tweaks to those methods?

Reply:
I think this function is the closest to `ReadRuntimeSystem` than `EmitRuntimeSource` since we don't emit the run-time inside the generated code. However, I think `ReadRuntimeSystem` is a terrible name since it essentially copies the runtime into a specific location.
I'll rename both Rust and Java version to `ImportRuntimeTo`

Reply:
I started also looking into merging them but this function does not lies in the same architecture as the SinglePassCompiler. There is another compiler in between and linking them together would be more verbose than necessary. Moreover, I see that the code significantly differ so I'm not sure tweaking the existing ReadRuntimeSystem would actually help anything. I'll skip that part as, even if we end up to want it, would probably require a PR of its own.